src/client/components/input/partial-date.js

```diff
@@ -25,7 +25,19 @@ const ymdRegex = /^\d{4}-\d{2}-\d{2}$/;
 const ymRegex = /^\d{4}-\d{2}$/;
 const yRegex = /^\d{4}$/;
 
+/**
+* This class is derived from the React Component base class
+* to render the partial date of any of the formats "YYYY", "YYYY-MM",
```

Comment:
This makes more sense if you put it as follows:

    ...and renders a partial date (any of the formats "YYYY",
    "YYYY-MM", "YYYY-MM-DD"). This allows...